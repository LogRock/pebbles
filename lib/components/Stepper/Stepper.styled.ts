import styled from "styled-components";

export const StepIndex = styled.div<{
  isCurStep: Boolean;
  isLastItem: Boolean;
  isFinished: Boolean;
}>`
  display: flex;
  position: relative;
  align-items: center;
  justify-content: center;
  width: 36px;
  height: 36px;
  margin-right: ${({ theme }) => theme.spacings.md};

  border: 2px solid
    ${(props: any) =>
      props.isCurStep || props.isFinished
        ? props.theme.colors.primary[400]
        : props.theme.colors.neutral[200]};
  border-radius: 96px;

  background-color: ${(props: any) =>
    props.isFinished ? props.theme.colors.primary[500] : "transparent"};
  box-shadow: 0px 0px 0px 4px
    ${(props: any) =>
      props.isCurStep ? props.theme.colors.primary[100] : "transparent"};

  p {
    display: ${(props: any) => (props.isFinished ? "none" : "block")};

    color: ${(props: any) =>
      props.isCurStep
        ? props.theme.colors.neutral[900]
        : props.theme.colors.neutral[500]};
    font-weight: 500;
  }

  &:after {
    content: "";

    display: ${(props: any) => (props.isLastItem ? "none" : "block")};
    position: absolute;
    top: 38px;
    left: 50%;
    height: 32px;

    transform: translate(-50%, 0);

    border: 1px solid
      ${(props: any) =>
        props.isFinished
          ? props.theme.colors.primary[500]
          : props.theme.colors.neutral[200]};
  }
`;

export const StepCheckMark = styled.div<{
  isCurStep: Boolean;
  isFinished: Boolean;
}>`
  display: ${(props: any) =>
    props.isFinished && !props.isCurStep ? "block" : "none"};
  position: relative;

  transform: translate(0, 0) rotate(45deg);

  &:after {
    content: "";

    position: absolute;
    width: 3px;
    height: 15px;

    transform: translate(0, -10px);

    border-radius: 8px;

    background-color: ${({ theme }) => theme.colors.neutral[900]};
  }

  &:before {
    content: "";

    position: absolute;
    width: 3px;
    height: 9px;

    transform: translate(-3px, 0) rotate(90deg);

    border-radius: 8px;

    background-color: ${({ theme }) => theme.colors.neutral[900]};
  }
`;

export const StepLabel = styled.div<{ isCurStep: Boolean }>`
  color: ${(props: any) =>
    props.isCurStep
      ? props.theme.colors.neutral[900]
      : props.theme.colors.neutral[500]};
  font-weight: 500;
`;

export const Step = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 34px;

  cursor: pointer;

  &:hover > ${StepLabel} {
    color: ${({ theme }) => theme.colors.neutral[900]};
  }

  &:hover > ${StepIndex} {
    border: 2px solid ${({ theme }) => theme.colors.primary[400]};

    p {
      color: ${({ theme }) => theme.colors.neutral[900]};
    }
  }
`;

export const Steps = styled.div`
  display: flex;
  flex-direction: column;

  font-family: ${({ theme }) => theme.loader.fontFamily};
`;
