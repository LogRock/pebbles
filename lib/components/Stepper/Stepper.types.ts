export interface StepperProps {
  labels?: string[];
  finishedSteps?: number[];
  curStep?: number;
  onStepClick?: (stepNum: number) => void;
}
