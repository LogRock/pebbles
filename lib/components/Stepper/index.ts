export { default } from "./Stepper";
export { default as StepperStep } from "./StepperStep";

export type { StepperProps, StepperStepProps } from "./Stepper.types";
