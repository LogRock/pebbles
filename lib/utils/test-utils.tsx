/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
