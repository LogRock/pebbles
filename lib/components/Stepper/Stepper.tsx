import React, { FC } from "react";

import { Steps } from "./Stepper.styled";
import { StepperProps } from "./Stepper.types";

const Stepper: FC<StepperProps> = ({ showHoverEffect, children, ...props }) => {
  return (
    <Steps showHoverEffect={showHoverEffect} {...props}>
      {children}
    </Steps>
  );
};

export default Stepper;
