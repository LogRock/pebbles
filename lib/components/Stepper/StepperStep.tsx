import React, { FC } from "react";

import { StepperProps } from "./Stepper.types";
import {
  Step,
  StepIndicatorContainer,
  StepIndicator,
  StepCheckMark,
  StepContent,
  StepLine,
} from "./Stepper.styled";

const StepperStep: FC<StepperProps> = ({
  stepIndex,
  last = false,
  completed = false,
  current = false,
  onStepClick,
  showHoverEffect = true,
  children,
}) => {
  return (
    <Step
      key={`${stepIndex}`}
      showHoverEffect={showHoverEffect}
      onClick={() => {
        onStepClick && stepIndex && onStepClick(stepIndex);
      }}
    >
      <StepIndicatorContainer>
        <StepIndicator current={current} last={last} completed={completed}>
          <p>{stepIndex}</p>
          <StepCheckMark current={current} completed={completed} />
        </StepIndicator>
        {!last && <StepLine completed={completed} />}
      </StepIndicatorContainer>

      <StepContent current={current}>{children}</StepContent>
    </Step>
  );
};

export default StepperStep;
