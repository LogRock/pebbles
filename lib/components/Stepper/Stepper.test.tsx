import React from "react";
import { render } from "../../utils/test-utils";

import Stepper from "./Stepper";
import StepperStep from "./StepperStep";

describe("Stepper test", () => {
  it("renders stepper with some content", () => {
    expect(
      render(
        <Stepper>
          <StepperStep stepIndex="01">
            <h4>Personal</h4>
          </StepperStep>
          <StepperStep stepIndex="02">
            <h4>Company</h4>
          </StepperStep>
          <StepperStep stepIndex="03">
            <h4>ELD</h4>
          </StepperStep>
        </Stepper>
      ).container.firstChild
    ).toMatchSnapshot();
  });

  it("renders in display only mode", () => {
    expect(
      render(
        <Stepper showHoverEffect>
          <StepperStep stepIndex="01">
            <h4>Personal</h4>
          </StepperStep>
          <StepperStep stepIndex="02">
            <h4>Company</h4>
          </StepperStep>
          <StepperStep stepIndex="03">
            <h4>ELD</h4>
          </StepperStep>
        </Stepper>
      ).container.firstChild
    ).toMatchSnapshot();
  });
});
