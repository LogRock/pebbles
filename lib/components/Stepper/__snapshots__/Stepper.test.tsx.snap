// Vitest Snapshot v1

exports[`Stepper test > renders in display only mode 1`] = `
<div
  class="sc-gKPRtg buorzf"
>
  <div
    class="sc-jSUZER eeLiNk"
  >
    <div
      class="sc-bcXHqe emxCrP"
    >
      <div
        class="sc-gswNZR hkQOUM"
      >
        <p>
          01
        </p>
        <div
          class="sc-hLBbgP fNXyZO"
        />
      </div>
      <div
        class="sc-dkrFOg VnjMe"
      />
    </div>
    <div
      class="sc-eDvSVe fYPDZJ"
    >
      <h4>
        Personal
      </h4>
    </div>
  </div>
  <div
    class="sc-jSUZER eeLiNk"
  >
    <div
      class="sc-bcXHqe emxCrP"
    >
      <div
        class="sc-gswNZR hkQOUM"
      >
        <p>
          02
        </p>
        <div
          class="sc-hLBbgP fNXyZO"
        />
      </div>
      <div
        class="sc-dkrFOg VnjMe"
      />
    </div>
    <div
      class="sc-eDvSVe fYPDZJ"
    >
      <h4>
        Company
      </h4>
    </div>
  </div>
  <div
    class="sc-jSUZER eeLiNk"
  >
    <div
      class="sc-bcXHqe emxCrP"
    >
      <div
        class="sc-gswNZR hkQOUM"
      >
        <p>
          03
        </p>
        <div
          class="sc-hLBbgP fNXyZO"
        />
      </div>
      <div
        class="sc-dkrFOg VnjMe"
      />
    </div>
    <div
      class="sc-eDvSVe fYPDZJ"
    >
      <h4>
        ELD
      </h4>
    </div>
  </div>
</div>
`;

exports[`Stepper test > renders stepper with some content 1`] = `
<div
  class="sc-gKPRtg gHXbGX"
>
  <div
    class="sc-jSUZER eeLiNk"
  >
    <div
      class="sc-bcXHqe emxCrP"
    >
      <div
        class="sc-gswNZR hkQOUM"
      >
        <p>
          01
        </p>
        <div
          class="sc-hLBbgP fNXyZO"
        />
      </div>
      <div
        class="sc-dkrFOg VnjMe"
      />
    </div>
    <div
      class="sc-eDvSVe fYPDZJ"
    >
      <h4>
        Personal
      </h4>
    </div>
  </div>
  <div
    class="sc-jSUZER eeLiNk"
  >
    <div
      class="sc-bcXHqe emxCrP"
    >
      <div
        class="sc-gswNZR hkQOUM"
      >
        <p>
          02
        </p>
        <div
          class="sc-hLBbgP fNXyZO"
        />
      </div>
      <div
        class="sc-dkrFOg VnjMe"
      />
    </div>
    <div
      class="sc-eDvSVe fYPDZJ"
    >
      <h4>
        Company
      </h4>
    </div>
  </div>
  <div
    class="sc-jSUZER eeLiNk"
  >
    <div
      class="sc-bcXHqe emxCrP"
    >
      <div
        class="sc-gswNZR hkQOUM"
      >
        <p>
          03
        </p>
        <div
          class="sc-hLBbgP fNXyZO"
        />
      </div>
      <div
        class="sc-dkrFOg VnjMe"
      />
    </div>
    <div
      class="sc-eDvSVe fYPDZJ"
    >
      <h4>
        ELD
      </h4>
    </div>
  </div>
</div>
`;
